distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1543,196 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
+        // @formatter:on
+
+        checkExpectations(cometDClient1, 1, 2);
+        checkExpectations(cometDClient2, 1, 2);
+
+        assertThat(cometDClient1.getAllMessages()
+                .size(), is(3));
+        assertThat(cometDClient2.getAllMessages()
+                .size(), is(3));
+
+        validateNotificationsSuccess(cometDClient1, filename, resourceData);
+        validateNotificationsSuccess(cometDClient2, filename, resourceData);
+
+        validateActivitiesSuccess(cometDClient1, filename, resourceData);
+        validateActivitiesSuccess(cometDClient2, filename, resourceData);
+
+    }
+
+    @Test
+    public void testTwoUsersSameProductRetryFailure() throws Exception {
+
+        CometDClient cometDClient1 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+        CometDClient cometDClient2 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+
+        String filename = "product3.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        setupCswServer(metacardId, filename, resourceData);
+        setupCswServer(metacardId, filename, resourceData);
+
+        String restUrl = REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + metacardId
+                + "?transform=resource" + "&session=" + cometDClient1.getClientId();
+
+        // Verify that product retrieval fails from the csw stub server.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().statusCode(500).contentType("text/plain")
+                .body(containsString("cannot retrieve product"));
+        // @formatter:on
+
+        checkExpectations(cometDClient1, 4, 8);
+        checkExpectations(cometDClient2, 4, 8);
+
+        validateNotificationsFailure(cometDClient1, filename);
+        validateNotificationsFailure(cometDClient2, filename);
+
+        validateActivitiesFailure(cometDClient1, filename);
+        validateActivitiesFailure(cometDClient2, filename);
+
+    }
+
+    private void setupCswServer(String metacardId, String filename, String resourceData)
+            throws Exception {
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(200), 3));
```

Comment:
I think you need to rebase. The latest PR by @garrettfreibott will require this line to be slightly modified into a builder style of method call.